jspsych-uil-randomization.js

```diff
@@ -32,6 +32,113 @@ else
 
     uil.randomization = {};
 
+    /**
+     * Checks whether an item may be appended to the randomized items
+     *
+     * @param randomized {array.<object>} A pre-existing list of stimuli
+     * @param constraints {object} The constraints used to examine whether an
+     *                             item can be appended to randomized. The keys
+     *                             need to be a key in the item parameter and
+     *                             the value is the maximum number allowable
+     *                             subsequent item[key] with the same value.
+     * @param item {object} The object which is intended to be appended. The
+     *                      constraints determine whether it fits.
+     *
+     * @return true if the item may be appended to randomized without violating
+     *         the constraints false otherwise.
+     */
+    function allowPushItem (
+        randomized,
+        constraints,
+        item
+    ) {
+        for (const [key, max] of Object.entries(constraints)) {
+            if (max < 1) {
+                throw new RangeError(
+                    "max is < 1; it's impossible to fit less than one repeating items"
+                );
+            }
+            let value = item[key];
+            if (value === undefined) {
+                throw ReferenceError(
+                    `There is no key "${key}" in item "${item}"`
+                );
+            }
+
+            let selection = randomized.slice(-max);
+            selection.push(item);
+
+            let count = 0;
+            selection.forEach(stimulus => count += stimulus[key] === value);
+            if (count > max)
+                return false;
```

Comment:
I think some comments could help here; I was about to write something (stupid) till I finally got what you're doing here. (I though you were checking the whole list of chosen items, but then I realised you were slicing the selection first).

Something like:
before line 68: "Slice the given array such that we have the last `max` number of items"
before line 72: "Check if the amount of item types in the sliced array does not exceed the contstraint"

Reply:
Add some comments in [6a7ae08](https://github.com/UiL-OTS-labs/jspsych-uil-utils/pull/20/commits/6a7ae08047d2705dcc6a98163f2e8b99f9cde5f4)

Reply:
That's much clearer, thank you!